src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -195,7 +195,7 @@ Error MeshForwarder::ForwardDuaToBackboneLink(Message &aMessage, const Ip6::Addr
     VerifyOrExit(Get<BackboneRouter::Local>().IsPrimary() && Get<BackboneRouter::Leader>().IsDomainUnicast(aDst),
                  error = kErrorNoRoute);
 
-    IgnoreError(Get<AddressResolver>().Resolve(aDst, dstRloc16, /* aAllowAddressQuery */ false));
+    SuccessOrExit(Get<AddressResolver>().Lookup(aDst, dstRloc16));
```

Comment:
_Quick note on the change here_: 
Changed this to `SuccessOrExit()` since if the call fails, then `dstRloc16` is not set (we don't initialize this variable above this call), so then the `VerifyOrExit(dstRloc16 == Get<Mle::MleRouter>().GetRloc16())` potentially accesses an uninitialized variable.

Reply:
Good catch!

We used `IgnoreError` explicitly, so maybe the understanding was that `dstRloc16` should be set to `0xfffe` in the case of error, which unfortunately is not true.

Should exit with `error = kErrorNoRoute` ?

Reply:
Good idea on `kErroNoRoute`. I will change `Lookup()` to return the looked up rloc16 instead of error (and then require `aRloc16` as input param). We can then return `kInvalidShortAddr` when it is not found.